Dockerfile

```diff
@@ -9,6 +9,14 @@ ENV PIP_DEFAULT_TIMEOUT=100 \
     PIP_NO_CACHE_DIR=1 \
     POETRY_VERSION=1.1.12
 
+RUN apt-get update && apt-get upgrade -y && apt-get install make cmake gcc libssl-dev wget -y
+RUN wget https://github.com/libgit2/libgit2/archive/refs/tags/v1.4.0.tar.gz -O libgit2-1.4.0.tar.gz \
+    && tar xzf libgit2-1.4.0.tar.gz \
+    && cd libgit2-1.4.0/ \
+    && cmake . \
+    && make \
+    && make install
+RUN apt-get install libffi-dev -y
```

Comment:
I wouldn't spend more than 10 minutes on this, but I was grappling with the problem independently this morning and have two observations:

1. The error is due to being unable to install binary wheels into the container. This approach works but isn't strictly required.
2. My alternate solution was to bump pygit2 from ^1.9.0 to ^1.10.0 for python versions 3.7 and later. However, be careful because I found that when I rebuilt poetry.lock with a new poetry (1.2.2) all of the unit testing for python 3.6 (which is forced to use the old poetry 1.1.x) broke because of a change in behavior surrounding setuptools support between poetry 1.1 and poetry 1.2.

I cannot explain why 1.9.x no longer works, but perhaps the base container python version bumped and 1.9.x doesn't have the needed content any longer? What you have is okay and has the virtue of working, but it's faintly annoying that we have to do it in the first place. See #403 for more ranting on this subject.

Reply:
Ya, the build was failing for quite some time. This at least gets it to work.